src/client/datascience/data-viewing/jupyterVariableDataProvider.ts

```diff
@@ -0,0 +1,104 @@
+
+import { inject, injectable, named } from 'inversify';
+
+import { JSONObject } from '@phosphor/coreutils';
+
+import { Identifiers } from '../constants';
+import { IJupyterVariable, IJupyterVariableDataProvider, IJupyterVariables, INotebook } from '../types';
+import { DataViewerDependencyService } from './dataViewerDependencyService';
+import { ColumnType, IDataFrameInfo } from './types';
+
+@injectable()
+export class JupyterVariableDataProvider implements IJupyterVariableDataProvider {
+    private initialized: boolean = false;
+
+    constructor(
+        @inject(IJupyterVariables) @named(Identifiers.ALL_VARIABLES) private variableManager: IJupyterVariables,
+        @inject(DataViewerDependencyService) private dependencyService: DataViewerDependencyService,
+        private variable: IJupyterVariable,
+        private notebook: INotebook
+    ) {}
+
+    /**
+     * Normalizes column types to the types the UI component understands.
+     * Defaults to 'string'.
+     * @param columns
+     * @returns Array of columns with normalized type
+     */
+    private static getNormalizedColumns(columns: { key: string; type: string }[]): { key: string; type: ColumnType }[] {
+        return columns.map((column: { key: string; type: string }) => {
+            let normalizedType: ColumnType;
+            switch (column.type) {
+                case 'bool':
+                    normalizedType = ColumnType.Bool;
+                    break;
+                case 'integer':
+                case 'int32':
+                case 'int64':
+                case 'float':
+                case 'float32':
+                case 'float64':
+                case 'number':
+                    normalizedType = ColumnType.Number;
+                    break;
+                default:
+                    normalizedType = ColumnType.String;
+            }
+            return {
+                key: column.key,
+                type: normalizedType
+            };
+        });
+    }
+
+    public dispose(): void {
+        return;
+    }
+
+    public async getDataFrameInfo(): Promise<IDataFrameInfo> {
+        await this.ensureInitialized();
+        return {
+            columns: this.variable.columns
+                ? JupyterVariableDataProvider.getNormalizedColumns(this.variable.columns)
+                : this.variable.columns,
+            indexColumn: this.variable.indexColumn,
+            rowCount: this.variable.rowCount || 0
+        };
+    }
+
+    public async getAllRows() {
+        let allRows: JSONObject = {};
+        await this.ensureInitialized();
+        if (this.variable.rowCount) {
+            allRows = await this.variableManager.getDataFrameRows(
+                this.variable,
+                this.notebook,
+                0,
+                this.variable.rowCount
+            );
+        }
+        return allRows;
+    }
+
+    public async getRows(start: number, end: number) {
+        let rows: JSONObject = {};
```

Comment:
As per other comments, please change to `any[]`.
```suggestion
        let rows: any[] = [];
```

Reply:
Sounds good to me, will do the change :) thanks!

Reply:
@DonJayamanne I have change it to:

`let rows: IRowsResponse = [];`

Where IRowsResponse is:

`export type IRowsResponse = any[];`